pygmt/src/basemap.py

```diff
@@ -39,6 +39,7 @@ def basemap(self, **kwargs):
     Full GMT docs at :gmt-docs:`basemap.html`.
 
     {aliases}
+       - J=projection
```

Comment:
As mentioned in https://github.com/GenericMappingTools/pygmt/issues/4011, with the new alias system, we have to manually maintain the list of aliases. Currently, the `J` alias is added at the end of the alias list. I guess we should sort the aliases alphabetically when we migrate more aliases into the new alias system.